client/file-browser/get-files.ts

```diff
@@ -13,34 +13,36 @@ function splitExtension(filename: string): [base: string, ext: string] {
 }
 
 export default async function readFolder(folderPath: string): Promise<FileBrowserEntry[]> {
-  try {
-    const entries = (await ipcRenderer.invoke('fsReadDir', folderPath, { withFileTypes: true }))!
-    return await Promise.all(
-      entries.map(async entry => {
-        const isFolder = entry.isDirectory
-        const [name, extension] = isFolder ? [entry.name, ''] : splitExtension(entry.name)
-        const stats = await ipcRenderer.invoke('fsStat', folderPath + '\\' + entry.name)
-        const path = folderPath + '\\' + entry.name
-
-        return isFolder
-          ? {
-              type: FileBrowserEntryType.Folder,
-              name,
-              path,
-            }
-          : {
-              type: FileBrowserEntryType.File,
-              name,
-              path,
-              extension: extension.toLowerCase(),
-              date: stats?.mtime ?? new Date(0),
-            }
-      }),
-    )
-  } catch (err) {
-    if ((err as any).code === 'ENOENT') {
-      throw new Error('Filepath ' + folderPath + " doesn't exist")
-    }
-    throw err
+  const readDirResult = (await ipcRenderer.invoke('fsReadDir', folderPath, {
+    withFileTypes: true,
+  }))!
+  if (readDirResult.error && readDirResult.code !== FsErrorCode.ENOENT) {
+    throw new Error('Unexpected FsError: ' + readDirResult.error)
   }
+  const entries: FsDirent[] = readDirResult.error ? [] : readDirResult.files
```

Comment:
If you do the above thing then you won't need to check for the error here anymore and this can simply be:
```ts
const entries = readDirResult.files
```
Or you can delete the line altogether then and just use the full name when doing the mapping.

Also, now that I think about it, it might make more sense to name the property inside the `FsReadDirResult` `entries`, instead of `files`, since that API returns both folders and files.